tests/suites/test_suite_nist_kw.function

```diff
@@ -1,159 +1,159 @@
 /* BEGIN_HEADER */
 #include "mbedtls/nist_kw.h"
+static int setup_psa_key(int usage,
+                         int alg,
+                         int key_type,
+                         mbedtls_svc_key_id_t *key_id,
+                         const data_t *key)
+{
+    int ret = PSA_SUCCESS;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+    ret = psa_crypto_init();
+    if (ret == PSA_SUCCESS) {
+        psa_set_key_usage_flags(&attributes, usage);
+        psa_set_key_algorithm(&attributes, alg);
+        psa_set_key_type(&attributes, key_type);
+        ret = psa_import_key(&attributes, key->x, key->len, key_id);
+    }
+
+    return ret;
+}
+
 /* END_HEADER */
 
 /* BEGIN_DEPENDENCIES
  * depends_on:MBEDTLS_NIST_KW_C
  * END_DEPENDENCIES
  */
 
-/* BEGIN_CASE depends_on:MBEDTLS_SELF_TEST:MBEDTLS_AES_C */
-void mbedtls_nist_kw_self_test()
+/*
+ * Test vectors taken from NIST
+ * https://csrc.nist.gov/Projects/Cryptographic-Algorithm-Validation-Program/CAVP-TESTING-BLOCK-CIPHER-MODES#KW
+ */
+/* BEGIN_CASE */
+void mbedtls_nist_kw_self_test(int mode, data_t *kw_key,
+                               data_t *kw_msg, data_t *kw_res, int kw_out_len)
 {
-    TEST_ASSERT(mbedtls_nist_kw_self_test(1) == 0);
+    unsigned char out[48];
```

Comment:
Rather than hard-coding a size, we should allocate an output buffer of exactly the expected size. This way, we can validate that the tested function doesn't insist on having a larger output buffer.

```
unsigned char out = NULL;
TEST_CALLOC(out, out_len);
```
and `mbedtls_free(out)` in the cleanup code.

(Sometimes we also test that the function works correctly when the output buffer is larger than needed, but we only do it when a white-box approach suggests that it might be a concern, which is not the case here.)

When we test with Asan (which we often do), it validates that the code doesn't try to access beyond the end of the buffer.